app/models/miq_request_workflow.rb

```diff
@@ -1042,6 +1042,9 @@ def allowed_storages(_options = {})
     storages = hosts.each_with_object({}) do |host, hash|
       host.writable_storages.each { |s| hash[s.id] = s }
     end.values
+
+    storages.reject!(&:maintenance)
```

Comment:
I agree with @agrare .  Also, I think available_ is the common method for other things (like hosts, and clusters), so naming wise it feels better to make a new method.  Although, @agrare, sounds like you'll have to update the automate best-fit logic as well.